documents: improve editor

sebastiendeleze opened the pull request:
Improve document editor by making some changes on fields, listed below.

* Adds CSS class `editor-title` to display a block for each main fields.
* Makes field `title.type` not editable and place it after the value.
* Removes type `bf:Local` for `contribution` field.
* Makes field `classification.type` not editable and place it after the value.
* Shows value when a `subtitle` field is added.
* Makes `language` field required and `language.type` readonly.
* Makes `partof` field required when document type is a contribution to journal.
* Makes `organisation` field required.
* Makes `provisionActivity` field required depending on document type, instead of hiding it.
* Closes #230.

Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

pronguen commented:
For type "Contribution to journal" (c_3e5a), subfields are missing and impossible to add to `partOf` (`numberingVolume`, `numberingIssue`). There should be the same subfields than for "Scientific article".

![image](https://user-images.githubusercontent.com/8154915/91410761-b11cd400-e847-11ea-8e2a-ab1c7dfcb179.png)$

There are 2 trashes for the field `editionStatement`

![image](https://user-images.githubusercontent.com/8154915/91411394-8ed78600-e848-11ea-83b6-1c3b63480463.png)

What is the base date field (so that we can build the year facet)? I think it is `provisionActivity.startDate` (if type=bf:Publication). This should be required in any document... I am sorry, spec were not clear. If we display `provisionActivity.startDate` all the time (as specified in https://github.com/rero/sonar/issues/254), this field must be required, right?

sebastiendeleze commented:
> For type "Contribution to journal" (c_3e5a), subfields are missing and impossible to add to `partOf` (`numberingVolume`, `numberingIssue`). There should be the same subfields than for "Scientific article".

Corrected.

> There are 2 trashes for the field `editionStatement`

Corrected.
 
> What is the base date field (so that we can build the year facet)? I think it is `provisionActivity.startDate` (if type=bf:Publication). This should be required in any document... I am sorry, spec were not clear. If we display `provisionActivity.startDate` all the time (as specified in #254), this field must be required, right?

Yes this is the provisionActivity.startDate which is taken for populating the year facet. Ok, by now, this field is required all the time.

> For a book chapter, I should be able to add a startDate in the `partOf`, but the select option is empty.

That's an issue with the editor. I created https://github.com/rero/ng-core/issues/247 for @jma.

mmo commented:
Seems ok, but not thoroughly tested.
